config/client.go

```diff
@@ -82,19 +82,18 @@ func (c *Config) NewESClient() (*http.Client, error) {
 	}
 
 	// Load CA certificate
-	caCert, err := ioutil.ReadFile(c.Elasticsearch.Client.CACert)
+	caCert, err := os.ReadFile(c.Elasticsearch.Client.CACert)
 	if err != nil {
 		return nil, xerrors.Errorf("error reading CA certificate file: %w", err)
 	}
 	caCertPool := x509.NewCertPool()
 	caCertPool.AppendCertsFromPEM(caCert)
 
-	tlsConfig := &tls.Config{ // nolint: gosec
+	tlsConfig := &tls.Config{ //nolint:gosec
 		Certificates: []tls.Certificate{cert},
 		RootCAs:      caCertPool,
 		ServerName:   c.Elasticsearch.Client.ServerName,
 	}
-	tlsConfig.BuildNameToCertificate()
```

Comment:
@dbellinghoven it would mean a great deal if you could confirm whether removing this makes sense